src/common/init.cpp

```diff
@@ -62,6 +64,35 @@ std::optional<ConfigError> InitConfig(ArgsManager& args, SettingsAbortFn setting
             fs::create_directories(net_path / "wallets");
         }
 
+        // Warn if we are trying to put the datadir on an exFAT fs on MacOS
+        // This is an upstream issue known to cause bugs, see #28552
+#ifdef __APPLE__
+        struct PathCheck {
+            fs::path path;
+            std::string_view description;
+        };
+        std::array<PathCheck, 2> paths_to_check{{
+            {args.GetDataDirNet(), "data directory"},
+            {args.GetBlocksDirPath(), "blocks directory"}
+        }};
+        for (const auto& check : paths_to_check) {
+            FSType fs_type = GetFilesystemType(check.path);
+            switch(fs_type) {
+                case FSType::EXFAT:
+                    InitWarning(strprintf(_("Specified %s \"%s\" is exFAT which is known to have intermittent corruption problems on MacOS. "
```

Comment:
For GUI, maybe we should check this while they still have the opportunity to change it...

Reply:
just one nit, the OS is named 'macOS'

Reply:
I'd be good to add a recommendation here, too, so the user is aware of possible immediate solutions.

```suggestion
                    InitWarning(strprintf(_("Specified %s \"%s\" is exFAT which is known to have intermittent corruption problems on MacOS, use APFS instead. "
```

Reply:
> For GUI, maybe we should check this while they still have the opportunity to change it...

To add this check to the *qt/intro.cpp* flow would require duplicating some of the init checks, as well as:

- first creating the chosen directory
- check what FS Type the created dir is
- if EXFAT prompt the user to change filesystems
- either remove the just-created directories (or keep them) depending on response.
- etc.

If we want that kind of logic then I'm happy to add it, but not sure if we need much more than a loud InitWarning myself.

> just one nit, the OS is named 'macOS'

Renamed in all places

Reply:
May be let's not insist so much and use ", consider using APFS instead."